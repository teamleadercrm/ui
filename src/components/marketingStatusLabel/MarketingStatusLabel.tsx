import React, { ReactNode } from 'react';
import Box from '../box';
import Icon from '../icon';
import cx from 'classnames';
import theme from './theme.css';
import { IconLockSmallFilled } from '@teamleader/ui-icons';
import { UITextBody, UITextSmall } from '../typography';
import { GenericComponent } from '../../@types/types';
import { BoxProps } from '../box/Box';
import { SIZES } from '../../constants';

interface MarketingStatusLabelProps extends Omit<BoxProps, 'ref'> {
  children?: ReactNode;
  fullWidth?: boolean;
  size?: Exclude<typeof SIZES[number], 'tiny' | 'large' | 'fullscreen' | 'smallest' | 'hero'>;
  className?: string;
}

const MarketingStatusLabel: GenericComponent<MarketingStatusLabelProps> = ({
  children,
  className,
  fullWidth = false,
  size = 'medium',
  ...others
}) => {
  const classNames = cx(theme['wrapper'], theme[`is-${size}`], className);

  const TextElement = size === 'small' ? UITextSmall : UITextBody;

  return (
    <Box
      {...others}
      alignItems="center"
      display={fullWidth ? 'flex' : 'inline-flex'}
      justifyContent="center"
      className={classNames}
      paddingHorizontal={2}
    >
      <TextElement className={theme['text']} marginRight={2}>
        {children}
      </TextElement>
      <Icon className={theme['icon']}>
        <IconLockSmallFilled />
      </Icon>
    </Box>
  );
};

export default MarketingStatusLabel;
