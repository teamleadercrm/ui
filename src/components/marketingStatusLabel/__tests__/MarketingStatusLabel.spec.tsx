import React from 'react';
import { render } from '@testing-library/react';
import MarketingStatusLabel from '..';

describe('Component - MarketingStatusLabel', () => {
  it('renders', async () => {
    const { asFragment } = render(<MarketingStatusLabel>Upgrade now</MarketingStatusLabel>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with an icon', async () => {
    const Icon = <span>Fake icon</span>;
    const screen = render(<MarketingStatusLabel icon={Icon}>Upgrade now</MarketingStatusLabel>);

    expect(screen.getByText('Fake icon')).toBeVisible();
    expect(screen.asFragment()).toMatchSnapshot();
  });
});
