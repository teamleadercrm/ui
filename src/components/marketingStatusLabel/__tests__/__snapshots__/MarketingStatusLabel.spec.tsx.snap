// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - MarketingStatusLabel renders 1`] = `
<DocumentFragment>
  <div
    class="box align-items-center display-inline-flex justify-content-center padding-left-2 padding-right-2 wrapper is-medium"
    data-teamleader-ui="marketing-status-label"
  >
    <p
      class="box margin-right-2 ui-text ui-text-body darkest text"
      data-teamleader-ui="ui-text"
    >
      Upgrade now
    </p>
    <span
      class="box align-items-center display-flex teal normal icon"
      data-teamleader-ui="icon"
    />
  </div>
</DocumentFragment>
`;

exports[`Component - MarketingStatusLabel renders with an icon 1`] = `
<DocumentFragment>
  <div
    class="box align-items-center display-inline-flex justify-content-center padding-left-2 padding-right-2 wrapper is-medium"
    data-teamleader-ui="marketing-status-label"
  >
    <p
      class="box margin-right-2 ui-text ui-text-body darkest text"
      data-teamleader-ui="ui-text"
    >
      Upgrade now
    </p>
    <span
      class="box align-items-center display-flex teal normal icon"
      data-teamleader-ui="icon"
    >
      <span
        opacity="0.84"
      >
        Fake icon
      </span>
    </span>
  </div>
</DocumentFragment>
`;
